src/eko/output/struct.py

```diff
@@ -0,0 +1,502 @@
+    interpolation_polynomial_degree: int
+    interpolation_is_log: bool
+    backward_inversion: Literal["exact", "expanded"]
+    n_integration_cores: int = 1
+
+
+@dataclass
+class Rotations(DictLike):
+    targetgrid: Optional[np.ndarray] = None
+    inputgrid: Optional[np.ndarray] = None
+    targetpids: Optional[np.ndarray] = None
+    inputpids: Optional[np.ndarray] = None
+
+    @classmethod
+    def default(cls, xgrid: interpolation.XGrid, pids: np.ndarray):
+        return cls(
+            targetgrid=xgrid.raw, inputgrid=xgrid.raw, targetpids=pids, inputpids=pids
+        )
+
+
+@dataclass
+class EKO:
+    """Operator interface.
+
+    This class offers an interface to an abstract operator, between memory and
+    disk.
+
+    An actual operator might be arbitrarily huge, and in particular size
+    limitations in memory are far more strict than on disk.
+    Since manually managing, for each application, the burden of off-loading
+    part of the operator might be hard and occasionally not possible (without a
+    clear picture of the internals), the library itself offers this facility.
+
+    In particular, the data format on disk has a complete specification, and
+    can hold a full operator independently of the loading procedure.
+    In order to accomplish the former goal, the remaining task of partial
+    loading is done by this class (for the Python library, other
+    implementations are possible and encouraged).
+
+    For this reason, a core component of an :cls:`EKO` object is a path,
+    referring to the location on disk of the corresponding operator.
+    Any :cls:`EKO` has an associated path:
+
+    - for the computed object, it corresponds to the path where the actual
+    result of the computation is already saved
+    - for a new object, it is the path at which any result of final or
+    intermediate computation is stored, as soon as it is produced
+
+    The computation can be stopped at any time, without the loss of any of the
+    intermediate results.
+
+    Attributes
+    ----------
+    path : pathlib.Path
+        path on disk, to which this object is linked (and for which it is
+        essentially an interface)
+    Q02 : float
+        inital scale
+    xgrid : interpolation.XGrid
+        momentum fraction internal grid
+    pids : np.ndarray
+        array of integers, corresponding to internal PIDs
+    configs : Configs
+        specific configuration to be used during the calculation of these
+        operators
+    rotations : Rotations
+        manipulation information, describing the current status of the EKO (e.g.
+        `inputgrid` and `targetgrid`)
+    debug : Debug
+        debug configurations
+    version : str
+        library version used to create the corresponding file
+    data_version : str
+        specs version, to which the file adheres
+
+    """
+
+    # operators cache, contains the Q2 grid information
+    _operators: Dict[float, Optional[Operator]]
```

Comment:
- `Dict` [is deprected](https://docs.python.org/3/library/typing.html#typing.Dict)
- making the value type optional seems wrong ...

Reply:
I know, is deprecated, but only since 3.9. I need this to support 3.8.
It is not, as written somewhere else, the in-memory operator **is** optional. Because it might simply be on disk.

Reply:
agreed